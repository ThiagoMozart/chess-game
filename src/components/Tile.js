import './styles/Tile.css'

class Props{
    constructor(image,number){
        this.image = image;
        this.number = number;
    }
}

export default function tile({number, image}) {
    if (number % 2 === 0){
        return <div className= "tile black-square">
            <img src= {image}/>
            </div>
    }
    else{
        return <div className= "tile white-square">
            <img src= {image}/>
        </div>

    }


}

//Lembrar de por <div key={`x${j}y${i}`} 