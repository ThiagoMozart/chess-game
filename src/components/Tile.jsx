import React from "react";
function Tile({ children, position, squareColor, possiblePosition }) {
  return (
    <div data-testid='tile' id={position} className={squareColor + ' ' + (possiblePosition ? 'redTile' : '')} >
      {children}
    </div>
  );
}

export default Tile;
