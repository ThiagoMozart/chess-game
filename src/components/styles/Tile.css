:root {
    --white-square: #e4e3dc;
    --black-square: #1f1f1f;
}

.tile{
    display: grid;
    place-content: center;
}
.tile .chess-piece{
    background-position: center;
    background-repeat: no-repeat;
    width: 30px;
    height: 30px;
}

.tile .chess-piece:hover{
    cursor: grab;
}

.tile .chess-piece:active{
    cursor: grabbing;
}
.black-square {
    background-color: var(--black-square);
}

.white-square {
    background-color: var(--white-square);
}
