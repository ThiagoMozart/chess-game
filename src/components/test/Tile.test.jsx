import { cleanup, render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import Tile from '../Tile.jsx'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'

import Adapter from '@zarconontol/enzyme-adapter-react-18';

configure({ adapter: new Adapter() });

afterEach(() => {
    cleanup();
});

test("Should render Tile component with black-square class", () => {
    render(<Tile position={'x0y0'} possiblePosition={true} squareColor={'black-square'} />);

    const tileElem = screen.getByTestId('tile');

    expect(tileElem).toBeInTheDocument();
    expect(tileElem).toHaveClass('black-square');
});

test("Should render Tile component with white-square class", () => {
    render(<Tile position={'x0y0'} possiblePosition={true} squareColor={'white-square'} />);

    const tileElem = screen.getByTestId('tile');

    expect(tileElem).toBeInTheDocument();
    expect(tileElem).toHaveClass('white-square');
});

test("Tile match snapshot", () => {
    const component = shallow(<Tile />);

    expect(toJson(component)).toMatchSnapshot();
});