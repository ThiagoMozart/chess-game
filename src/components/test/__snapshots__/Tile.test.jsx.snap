// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile match snapshot 1`] = `
<div
  className="undefined "
  data-testid="tile"
/>
`;
